### PR TITLE
[xlnt] Upgrade to v1.4.0

### PR DESCRIPTION
**Pull request by cnbatch**
**Describe the pull request**

- What does your PR fix? Fixes issue #
Upgrade xlnt to version 1.4.0

- Which triplets are supported/not supported? Have you updated the CI baseline?
All supported.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

File: ports/xlnt/portfile.cmake [hunk 1]
```diff
@@ -1,30 +1,41 @@
...
+vcpkg_install_cmake()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows" OR (NOT CMAKE_SYSTEM_NAME AND CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows"))
+    if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
+        vcpkg_apply_patches(
+            SOURCE_PATH ${CURRENT_PACKAGES_DIR}
+            PATCHES ${CMAKE_CURRENT_LIST_DIR}/static-linking-for-windows.patch
+        )
+    endif()
+endif()
```
Comment (PhoebeHui) [inline ports/xlnt/portfile.cmake]: This is deprecated, could you please update it? 
 
```
if(VCPKG_TARGET_IS_WINDOWS)
     set(win_patch "static-linking-for-windows.patch")
endif()

vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO tfussell/xlnt
    REF v1.4.0
    SHA512 74abbee97994098fb7d8fd0839929db74fe01b8428f8bdb8edd28340d3b3ed04d4c7d6dd5d886ae766054ff1b0fe9a8275098a1462e7a5146ff09f1cdb063360
    HEAD_REF master
    PATCHES
        ${win_patch}
)
```

File: ports/xlnt/portfile.cmake [hunk 2]
```diff
@@ -1,30 +1,41 @@
...
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/xlnt)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/share/man)
+file(INSTALL ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/xlnt RENAME copyright)
```
Comment (PhoebeHui) [inline ports/xlnt/portfile.cmake]: We usually use 'file(INSTALL ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)' instead.

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/xlnt/portfile.cmake (hunk 2 above)
Comment (cnbatch) [reply ports/xlnt/portfile.cmake]: Thanks for reviewing. New changes are made to portfile.cmake file.

**Review by PhoebeHui — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**